From c2835fca772990acfd008cc1a5c67ffe646148b5 Mon Sep 17 00:00:00 2001
From: Ivan Mironov <mironov.ivan@gmail.com>
Date: Tue, 17 Feb 2026 16:57:43 +0100
Subject: [PATCH] acp: Use musl build of codex-acp on Linux

This allows codex-acp to work inside devcontainers with older versions
of libc or with other shared libraries missing (like libssl.so.3).

Normal dynamic+libc build fails with following error in the log file:

	2026-02-16T05:31:20+01:00 WARN  [agent_servers::acp] agent stderr: /root/.local/share/zed/external_agents/codex/v0.9.2/codex-acp: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /root/.local/share/zed/external_agents/codex/v0.9.2/codex-acp)

And GUI just shows infinite "Loading..." in this case.

Fixes: https://github.com/zed-industries/zed/issues/45927
---
 crates/project/src/agent_server_store.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crates/project/src/agent_server_store.rs b/crates/project/src/agent_server_store.rs
index 3dcbab8bafb4..fbbcb2cbf306 100644
--- a/crates/project/src/agent_server_store.rs
+++ b/crates/project/src/agent_server_store.rs
@@ -1689,7 +1689,7 @@ fn get_platform_info() -> Option<(&'static str, &'static str, &'static str)> {
     } else if cfg!(target_os = "windows") {
         "pc-windows-msvc"
     } else if cfg!(target_os = "linux") {
-        "unknown-linux-gnu"
+        "unknown-linux-musl"
     } else {
         return None;
     };
-- 
2.53.0

